Realm/Realm/Sync/Session.cs

```diff
@@ -27,21 +30,76 @@ namespace Realms.Sync
     /// and the server (and a remote Realm served by a MongoDB Realm Server). Sessions are always created by the SDK and vended
     /// out through various APIs. The lifespans of sessions associated with Realms are managed automatically.
     /// </summary>
-    public class Session
+    public class Session : INotifyPropertyChanged
     {
         /// <summary>
         /// Triggered when an error occurs on a session. The <c>sender</c> argument will be the session which has errored.
         /// </summary>
         public static event EventHandler<ErrorEventArgs> Error;
 
+        private readonly SessionHandle _handle;
+
+        [SuppressMessage("StyleCop.CSharp.NamingRules", "SA1300:Element should begin with upper-case letter", Justification = "This is the private event - the public is uppercased.")]
+        private event PropertyChangedEventHandler _propertyChanged;
+
+        private IDisposable _propertyChangedNotificationTokens;
```

Comment:
```suggestion
        private IDisposable _notificationToken;
```

There's no need for `Session` to know how notifications are implemented in Core and that there are multiple tokens.